lighthouse-core/audits/autocomplete.js

```diff
@@ -43,30 +185,83 @@ class AutocompleteAudit extends Audit {
     };
   }
 
+  /**
+   * @param {LH.Artifacts.FormInput} input
+   * @return {{isValid: Boolean, orderWarn?: Boolean}}
+   */
+  static isValidAutocomplete(input) {
+    if (!input.autocomplete.attribute) return {isValid: false};
+    if (input.autocomplete.attribute.includes(' ') ) {
+      const tokenArray = input.autocomplete.attribute.split(' ');
+      for (const token of tokenArray) {
+        if (token.slice(0, 8) === 'section-') continue;
+        if (validAutocompleteTokens.includes(token)) continue;
+        return {isValid: false};
+      }
+      if (!input.autocomplete.property) return {isValid: false, orderWarn: true};
+      return {isValid: true};
+    }
+    return {isValid: validAutocompleteTokens.includes(input.autocomplete.attribute)};
+  }
+
   /**
    * @param {LH.Artifacts} artifacts
    * @return {LH.Audit.Product}
    */
   static audit(artifacts) {
     const forms = artifacts.FormElements;
     const failingFormsData = [];
+    const warnings = [];
+    let notApplicable = 0;
+    let inputsCount = 0;
     for (const form of forms) {
       for (const input of form.inputs) {
-        if (!input.autocomplete) {
-          failingFormsData.push({
-            node: /** @type {LH.Audit.Details.NodeValue} */ ({
-              type: 'node',
-              snippet: input.snippet,
-              nodeLabel: input.nodeLabel,
-            }),
-          });
+        inputsCount += 1;
+        const token = this.isValidAutocomplete(input);
+        if (!token.isValid) {
+          if (!input.autocomplete.prediction) {
```

Comment:
this sequence of `if` statements would be stronger if it's strung together as a larger if/else sequence.   right now there are some else cases that are falling through the cracks that we care about.

Reply:
which cases do you think should be caught in the else statement?

Reply:
@paulirish